File: CTCostomerBusiness/CTCommon/src/main/java/com/maroon/ct/common/util/PhoenixUtil.java
package com.maroon.ct.common.util;

import com.alibaba.fastjson.JSONObject;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;

/**
 * 通过phoenix操作hbase的工具类
 */
public class PhoenixUtil {

    private Connection conn;

    public PhoenixUtil(Connection conn) {
        this.conn = conn;
    }

    public void setConn(Connection conn) {
        this.conn = conn;
    }

    /**
     * 将Hbase中查询的结果返回成json对象串列表
     * @param sql 查询的sql语句
     * @return
     * @throws SQLException
     */
    public List<JSONObject> queryList(String sql) throws SQLException {
        // 创建数据库操作对象
        PreparedStatement ps = conn.prepareStatement(sql);
        // 执行sql语句
        ResultSet resultSet = ps.executeQuery();                    // 数据
        ResultSetMetaData resMetaData = resultSet.getMetaData();    // 数据元信息
        // 结果列表
        List<JSONObject> resListJson = new ArrayList<JSONObject>();
        while (resultSet.next()) {
            // 创建Json对象
            JSONObject jsonObject = new JSONObject();
            // 将结果封装到Json对象中
            for (int i = 0; i < resMetaData.getColumnCount(); i++) {
                jsonObject.put(resMetaData.getTableName(i), resultSet.getObject(i));
            }
            // 结果添加到结果列表中
            resListJson.add(jsonObject);
        }
        // 释放资源
        resultSet.close();
        ps.close();
        conn.close();

        return resListJson;
    }


}

File: CTCostomerBusiness/CTConsumer/src/main/scala/com/maroon/ct/consumer/util/PhoenixUtil.scala
package com.maroon.ct.consumer.util

import com.alibaba.fastjson.JSONObject
import java.sql.{Connection, DriverManager, PreparedStatement, ResultSet}
import scala.collection.mutable.ListBuffer

object PhoenixUtil {

  var propertiesName: String = _ // 配置文件平常
  var conn: Connection = _
  var resultSet: ResultSet = _
  var ps: PreparedStatement = _

  /**
   * 开启资源
   * @return
   */
  def open(): Connection = { // 获取到连接
    val properties = PropertiesUtil.load(propertiesName) // 加载配置文件参数
    Class.forName(properties.getProperty("phDriver")) // 驱动类phoenixDriver
    val conn: Connection = DriverManager.getConnection(properties.getProperty("phUrl")) // 获取连接配置
    conn
  }

  /**
   * 释放资源
   */
  def close(): Unit ={
    resultSet.close()
    ps.close()
    conn.close()
  }

  /*
  将Hbase中查询的结果返回成json对象串列表
   */
  def queryList(sql: String): List[JSONObject] = {
    conn = open()
    // 创建数据库操作对象
    ps = conn.prepareStatement(sql)
    // 执行sql语句
    resultSet = ps.executeQuery()
    val resMetaData = resultSet.getMetaData // 获取数据的元数据信息
    // 处理结果集
    val listBufferJson = new ListBuffer[JSONObject]()
    while (resultSet.next()) {
      // 创建JSON对象
      val jsonObject = new JSONObject()
      // 将结果封装成Json对象
      //{"erpcode":"xxx","if_new_material":"1"}
      for (i <- 1 to resMetaData.getColumnCount) {
        jsonObject.put(resMetaData.getColumnName(i), resultSet.getObject(i))
      }
      listBufferJson.append(jsonObject)
    }
    // 释放资源
    close()
    // 结果输出
    listBufferJson.toList
  }
}
